src/Options.php

```diff
@@ -0,0 +1,307 @@
+<?php
+
+namespace Graby\HttpClient\Plugin\ServerSideRequestForgeryProtection;
+
+use Graby\HttpClient\Plugin\ServerSideRequestForgeryProtection\Exception\InvalidOptionException;
+
+class Options
+{
+    /**
+     * Allow credentials in a URL
+     *
+     * @var bool
+     */
+    private $sendCredentials = false;
+
+    /**
+     * Pin DNS records
+     *
+     * @var bool
+     */
+    private $pinDns = false;
+
+    /**
+     * blacklist and whitelist
+     *
+     * @var array
+     */
+    private $lists = [
+        'whitelist' => [
+            'ip' => [],
+            'port' => ['80', '443', '8080'],
+            'domain' => [],
+            'scheme' => ['http', 'https'],
+        ],
+        'blacklist' => [
+            'ip' => [
+                '0.0.0.0/8',
+                '10.0.0.0/8',
+                '100.64.0.0/10',
+                '127.0.0.0/8',
+                '169.254.0.0/16',
+                '172.16.0.0/12',
+                '192.0.0.0/29',
+                '192.0.2.0/24',
+                '192.88.99.0/24',
+                '192.168.0.0/16',
+                '198.18.0.0/15',
+                '198.51.100.0/24',
+                '203.0.113.0/24',
+                '224.0.0.0/4',
+                '240.0.0.0/4',
+            ],
+            'port' => [],
+            'domain' => [],
+            'scheme' => [],
+        ],
+    ];
+
+    /**
+     * Get send credentials option.
+     *
+     * @return bool
+     */
+    public function getSendCredentials()
+    {
+        return $this->sendCredentials;
+    }
+
+    /**
+     * Enable sending of credenitals
+     * This is potentially a security risk.
+     *
+     * @return Options
+     */
+    public function enableSendCredentials()
+    {
+        $this->sendCredentials = true;
+
+        return $this;
+    }
+
+    /**
+     * Disable sending of credentials.
+     *
+     * @return Options
+     */
+    public function disableSendCredentials()
+    {
+        $this->sendCredentials = false;
+
+        return $this;
+    }
+
+    /**
+     * Get pin DNS option.
+     *
+     * @return bool
+     */
+    public function getPinDns()
+    {
+        return $this->pinDns;
+    }
+
+    /**
+     * Enable DNS pinning.
+     *
+     * @return Options
+     */
+    public function enablePinDns()
+    {
+        $this->pinDns = true;
+
+        return $this;
+    }
+
+    /**
+     * Disable DNS pinning.
+     *
+     * @return Options
+     */
+    public function disablePinDns()
+    {
+        $this->pinDns = false;
+
+        return $this;
+    }
+
+    /**
+     * Checks if a specific value is in a list.
+     *
+     * @param 'whitelist'|'blacklist' $listName
```

Comment:
Wrong order for description & type.

See https://scrutinizer-ci.com/g/j0k3r/httplug-ssrf-plugin/inspections/b9d2643e-5c6a-4ffe-9c5f-31dcf0b966b0/issues/files/src/Options.php?status=new&orderField=path&order=asc&honorSelectedPaths=0

Reply:
For me the order is good, but it does not exist a norm for a list of accepted values . I moved that in the description